### PR TITLE
Add some extensions

### PR DESCRIPTION
**Pull request by AlexDenisov**

### CODE REVIEW

**Review by cgestes — APPROVED** [review]
nice, please dont squash and *keep the commits* :)

I checked that the behavior of PyTorch is the same with those changes in.
Also checked for memory mistakes.

All seems good :)

File: torch/csrc/cuda/shared/nvtx.cpp [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
+struct RangeHandle {
+  nvtxRangeId_t id;
+  const char* msg;
+};
+
+static void flex_callback_range_end(void* userData) {
+  RangeHandle* handle = ((RangeHandle*)userData);
+  nvtxRangeEnd(handle->id);
+  free((void*)handle->msg);
+  free((void*)handle);
```
Comment (vincent-tr) [inline torch/csrc/cuda/shared/nvtx.cpp:19]: Very much of a detail, which may also depends on conventions that I don't know here:
if `msg` is not marked const in the structure definition above, you should not need the cast.
Unless I miss something, I also think the cast for `handle` itself is not necessary.

Another nitpicking question:
It looks like most of this file is using c++ features over c. Shouldn't we use new/delete and std::string here?

Comment (AlexDenisov) [reply torch/csrc/cuda/shared/nvtx.cpp:19]: C++ would complain without casts. `new`/`delete` would work just fine as well, shall I switch to it instead? I don't have a strong opinion on this one, it's just feels like doing it C-way is a better option for some reason 😅

Comment (AlexDenisov) [reply torch/csrc/cuda/shared/nvtx.cpp:19]: @vincent-tr I'll merge as is, let me know if you'd prefer switching to `new`/`delete`/`std::string`: we can do it any time. Alternatively, do not hesitate to submit a PR 😄

Comment (vincent-tr) [reply torch/csrc/cuda/shared/nvtx.cpp:19]: Was only nitpicking suggestions.
If you feel more comfortable using C-way, then just keep it.

**Pull request merged**